src/components/main-content.tsx

```diff
@@ -15,16 +15,14 @@ export class MainContentComponent extends BaseComponent<IProps, IState> {
   public render() {
     const {simulation} = this.stores;
     const primaryMag = simulation.getMagnetAtIndex(0);
-    const primaryMagType = primaryMag ? primaryMag.type : "none";
     const secondaryMag = simulation.getMagnetAtIndex(1);
-    const secondaryMagType = secondaryMag ? secondaryMag.type : "none";
     return (
       <div className="main-content">
         <MagnetCanvas
           width={800}
           height={500}
-          showMagnet1={primaryMagType !== "none"}
-          showMagnet2={secondaryMagType !== "none"}
+          showMagnet1={primaryMag !== null}
+          showMagnet2={secondaryMag !== null}
```

Comment:
As Kirk pointed out elsewhere, this won't catch `undefined`, so `magnet != null` or `!!magnet` is safer

Reply:
Fixed.